### DISCUSSION TITLE
ATTiny1634 avrdude: Yikes!  Invalid device signature.

### DISCUSSION THREAD

**Discussion by triodit** [SpenceKonde/ATTinyCore#582] [Support / Q & A]
So my current setup is an Arduinio Nano knockoff (Old Bootloader)  running arduino as ISP.  And I'm trying to use it to load code into a ATTINY1643R.  I've used this setup on an ATTiny85 without issues.  But I'm getting this error.  Anyone know what this might be???

Sketch uses 822 bytes (5%) of program storage space. Maximum is 16384 bytes.
Global variables use 9 bytes (0%) of dynamic memory, leaving 1015 bytes for local variables. Maximum is 1024 bytes.
avrdude: Yikes!  Invalid device signature.
         Double check connections and try again, or use -F to override
         this check.

An error occurred while uploading the sketch

![image](https://user-images.githubusercontent.com/82182510/121820070-8fa99f00-cc45-11eb-8b61-95596c7531be.png)

**Comment by triodit** [comment]
![20210613_124941](https://user-images.githubusercontent.com/82182510/121820155-0cd51400-cc46-11eb-9b21-c5eb82a97383.jpg)

**Comment by triodit** [comment]
sometimes I get this as well

 Sketch uses 822 bytes (5%) of program storage space. Maximum is 16384 bytes.
Global variables use 9 bytes (0%) of dynamic memory, leaving 1015 bytes for local variables. Maximum is 1024 bytes.
avrdude: Expected signature for ATtiny1634 is 1E 94 12
         Double check chip, or use -F to override this check.
Wrong microcontroller found.  Did you select the right board from the Tools > Board menu?

**Reply by prandeamus** [reply]
Using -F to ignore the signatures is nearly always a bad move, or so I've been told. Is it for corner cases on some chips where under very odd circumstances the sig gets corrupted? Anyway, I think you'll find it's rarely the right thing.

Have you tried slowing the bit rate using -B4 or -B10 or some larg-ish number after B? Sometimes you get bad signatures because the chip can't keep up, e.g. because fuse has been scrambled and is using v. slow clock. That's generic advice, as I've never used a 1634. More experienced folk may have better ideas.

**Comment by SpenceKonde** [comment]
Enable verbose output on upload so you can see what sig it is seeing. If it sees an m328p, it is getting the bootloader fpr the nano; disable auto reset with an approx. 10 uF cap between reset and ground. 

If it sees 0x000000 or something else wrong - is it a virgin chip? (ie, has it ever been bootloaded or is it fresh from factory)? If fresh from factory, suspect that SCK clock is too fast, the 1634, 841, and probably the 828 as well are known to be pickier about this than most classic AVRs are, so modify ArduinoasISP to slow it down (if you just start scrolling down and skimming the comments, there's a section where they talk about that; I suspect slowing it down would fix it. ArduinoasISP doesn't use the -B options

If not virgin, or if you were trying to bootload it and may not have gotten an error or not gotten same error the first time, could you have set it to use a crystal? In that case,it needs either a crystal connected (not gonna happen with that clip) or a clock signal to stand in place of the crystals output on the XTAL1 pin, which you can generate by taking over a timer on the 328p and using it to make like, a 2 MHz clock?

**Comment by triodit** [comment]
Ok so I rebuilt the whole jig and it worked.  NO IDEA what was wrong but thats fine!   I can move to figuring out this debugWIRE stuff now when my ICE programer gets here!  TY FOR THE HELP!
